hugolib/page_test.go

```diff
@@ -1059,6 +1113,9 @@ func TestMetadataDates(t *testing.T) {
 		checkDate(t, i+1, "LastMod", p.ExpiryDate, test.expExp, fi)
 
 		// check Page Params
+		// TODO(bep) we need to rewrite these date tests to more unit style.
+		// The params checks below are currently flawed, as they don't check for the
+		// absense (nil) of a date.
```

Comment:
I fixed this in PR #4340. Please see [my comment](https://github.com/gohugoio/hugo/pull/4436#issuecomment-367775422) on why you really should just merge it.

I'm not sure what you mean by "more unit style". I don't think you mean tests of smaller "units" such as an individual functions. What I did with these tests is to cover "the complex" that governs date values (front matter, filenames, file timestamps)... the "unit" is the blackbox the is Hugo's content processing. Or perhaps the table of inputs and outputs don't seem unit style to you? But that table is the data for data-driven unit tests. Inputs and outputs stripped of all the scaffolding code.

Reply:
A creating a site, then do a full build, and check the result will by most people categorised as an integration test. We do that a lot in Hugo, because building all is often cheaper/easier than creating testable and smaller units. My new `frontmatterHandler` has no dependencies to `Page` or `Site` and can be tested as _an unit_.